app/src/components/HomePageJN/HomePage.tsx

```diff
@@ -86,6 +95,13 @@ export default function HomePage({
       skip: !cityIdFromParam,
     });
 
+  // If city query fails (e.g., non-existing city ID), redirect to cities onboarding
+  useEffect(() => {
+    if (cityError) {
+      router.push(`/${lng}/cities`);
+    }
+  }, [cityError, cityIdFromParam, lng, router]);
```

Comment:
I think it might be good to not have these errors be caught silently but somehow reported to the user?